File: programmers/jihwjeon/42587.py
from collections import deque

def solution(priorities, location):
    answer = 0
    q = deque([])
    for i, p in enumerate(priorities):
        q.append([p, i])
        
    while q:
        if q[0][0] == max(q)[0]:
            answer += 1
            v, i = q.popleft()
            if i == location:
                break
        else:
            x = q.popleft()
            q.append(x)
                
    return answer

File: programmers/yuhwang/42587.swift
import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    var answer = 1
    var prior = priorities
    var location = location
    while prior.count > 0 {
        if prior[0] < prior.max()! {
            prior.append(prior[0])
        } else {
            if location == 0 { return answer }
            answer += 1
        }
        prior.removeFirst()
        location -= 1
        if location < 0 {
            location = prior.count - 1
        }
    }
    return answer
}